File: 020/Problem20.mjs
import { performance } from "perf_hooks";

function problem() {
  var startNum = 100;

  var prod = BigInt(1);
  for (var i = startNum; i > 1; i--) {
    prod *= BigInt(i);
  }
  const prodStr = prod.toString();
  var sum = 0;
  for (var i = 0; i < prodStr.length; i++) {
    sum += parseInt(prodStr[i]);
  }
  console.log(sum);
}

var start = performance.now();
for (var i = 0; i < 1000; i++) {
  problem();
}
var end = performance.now();
console.log("Function took " + (end - start));

File: 020/Problem20.go
package main

import (
	"fmt"
	"math/big"
	"strconv"
	"time"
)

func problem() {
	// Starting number
	startNum := int64(100)

	a := startNum
	prod := big.NewInt(1)
	for a > 1 {
		prod.Mul(prod, big.NewInt(a))
		a--
	}
	ansString := prod.String()
	sum := 0
	for i := 0; i < len(ansString); i++ {
		tmp, _ := strconv.Atoi(string(ansString[i]))
		sum += tmp
	}
	fmt.Println(sum)
}

func main() {
	numLoops := 1000
	start := time.Now()
	for i := 0; i < numLoops; i++ {
		problem()
	}
	elapsed := time.Since(start)
	fmt.Printf("Function took %s", elapsed)
}
